src/Foundatio/Storage/IFileStorage.cs

```diff
@@ -18,7 +19,61 @@ public interface IFileStorage : IHaveSerializer, IDisposable {
         Task<bool> CopyFileAsync(string path, string targetPath, CancellationToken cancellationToken = default);
         Task<bool> DeleteFileAsync(string path, CancellationToken cancellationToken = default);
         Task DeleteFilesAsync(string searchPattern = null, CancellationToken cancellation = default);
-        Task<IEnumerable<FileSpec>> GetFileListAsync(string searchPattern = null, int? limit = null, int? skip = null, CancellationToken cancellationToken = default);
+        Task<FileListResult> GetFileListAsync(string searchPattern = null, int? limit = null, CancellationToken cancellationToken = default);
+    }
+
+    public interface IHasNextPageFunc {
+        Func<Task<NextPageResult>> NextPageFunc { get; set; }
+    }
+
+    public class NextPageResult {
+        public bool Success { get; set; }
```

Comment:
Would be nice if these were readonly

Reply:
It’s internal to implementation and I need them to support modifying.